content/en/blog/_posts/2022-12-20-validating-admission-policies-alpha/index.md

```diff
@@ -0,0 +1,156 @@
+CustomResourceDefinitions](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules).
+This enhancement expands the use of CEL in Kubernetes to support a far wider
+range of admission use cases.
+
+Admission webhooks can be burdensome to develop and operate. Webhook developers
+must implement and maintain a webhook binary to handle admission requests. Also,
+admission webhooks are complex to operate. Each webhook must be deployed,
+monitored and have a well defined upgrade and rollback plan. To make matters
+worse, if a webhook times out or becomes unavailable, the Kubernetes control
+plane can become unavailable. This enhancement avoids much of this complexity of
+admission webhooks by embedding CEL expressions into Kubernetes resources
+instead of calling out to a remote webhook binary.
+
+For example, to set a limit on how many replicas a Deployment can have.
+Start by defining a validation policy:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "demo-policy.example.com"
+spec:
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+  validations:
+    - expression: "object.spec.replicas <= 5"
+```
+
+The `expression` field contains the CEL expression that is used to validate
+admission requests. `matchConstraints` declares what types of requests this
+`ValidatingAdmissionPolicy` is may validate.
+
+Next bind the policy to the appropriate resources:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "demo-binding-test.example.com"
+spec:
+  policy: "demo-policy.example.com"
+  matchResources:
+    namespaceSelector:
+    - key: environment,
+      operator: In,
+      values: ["test"]
+```
+
+This `ValidatingAdmissionPolicyBinding` resource binds the above policy only to
+namespaces where the `environment` label is set to `test`. Once this binding
+is created, the kube-apiserver will begin enforcing this admission policy.
+
+To emphasize how much simpler this approach is than admission webhooks, if this example
+were instead implemented with a webhook, an entire binary would need to be
+developed and maintained just to perform a `<=` check. In our review of a wide
+range of admission webhooks used in production, the vast majority performed
+relatively simple checks, all of which can easily be expressed using CEL.
+
+Validation admission policies are highly configurable, enabling policy authors
+to define policies that can be parameterized and scoped to resources as needed
+by cluster administrators.
+
+For example, the above admission policy can be modified to make it configurable:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "demo-policy.example.com"
+Spec:
```

Comment:
```suggestion
spec:
```

Reply:
@jpbetz I think this is the only point left before being able to approve/merge.

Reply:
I can (and will) use admin access to accept this change.